src/ops/ops.go

```diff
@@ -281,9 +281,23 @@ func (o *ops) renderDeploymentFiles(srcTemplate string, params map[string]string
 	o.log.Infof("Filling template file %s", srcTemplate)
 	tmpl := template.Must(template.New("assisted-controller").Parse(string(templateData)))
 	var buf bytes.Buffer
-	if err = tmpl.Execute(&buf, params); err != nil {
-		o.log.Errorf("Failed to render controller template: %e", err)
-		return err
+	var content []byte
+	if config.GlobalConfig.ServiceIPs != "" && strings.Contains(srcTemplate, controllerDeployPodTemplate) {
```

Comment:
I don't think this logic belongs here. This is a generic rendering function, can you move this to `renderControllerPod` as it will only ever apply to the pod case?

Reply:
I had thought of doing that but the difference, in this case, is also for the type of params argument sent to `renderDeploymentFiles()` i.e. for onprem, param needs to be of type ` map[string][]string` while originally its ` map[string]string`Are you thinking of adding a separate method just for onprem similar to `renderDeploymentFiles`? I took this approach as then most of the code can be used as is.

Reply:
Can you change `params` to be `map[string]interface{}` in this function?

Reply:
Then I don't think you need to change this function at all, right?

Reply:
Well, except for the parameter type change :sweat_smile: